packages/terra-search-field/src/SearchField.jsx

```diff
@@ -183,36 +225,57 @@ class SearchField extends React.Component {
       inputAttributes,
       ...customProps
     } = this.props;
+
+    const cancelButtonVisible = !defaultValue && !this.state.searchText ? 'input-clear' : undefined;
+
+    const cancelButtonClassNames = cx([
+      'clear',
+    ]);
+
+    const inputClassNames = cx([
+      'input',
+      cancelButtonVisible,
+    ]);
+
     const searchFieldClassNames = cx([
       'searchfield',
       { block: isBlock },
       customProps.className,
     ]);
 
-    const valueTemp = {};
-    if (value !== undefined) {
-      valueTemp.value = value;
-    } else {
-      valueTemp.defaultValue = defaultValue;
-    }
-
     const inputText = this.context.intl.formatMessage({ id: 'Terra.searchField.search' });
     const buttonText = this.context.intl.formatMessage({ id: 'Terra.searchField.submit-search' });
-    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes, valueTemp);
+    const additionalInputAttributes = Object.assign({ 'aria-label': inputText }, inputAttributes);
 
     return (
       <div {...customProps} className={searchFieldClassNames}>
         <Input
-          className={cx('input')}
+          defaultValue={defaultValue}
+          className={inputClassNames}
           type="search"
           placeholder={placeholder}
           onChange={this.handleTextChange}
+          onFocus={this.handleInputFocus}
           disabled={isDisabled}
           aria-disabled={isDisabled}
           onKeyDown={this.handleKeyDown}
           refCallback={inputRefCallback}
+          value={defaultValue ? undefined : this.state.searchText}
           {...additionalInputAttributes}
         />
+        { this.previouslyFocusedInput && this.state.searchText
+          ? (
+            <Button
+              className={cancelButtonClassNames}
+              onClick={this.handleClear}
+              text="cancel"
+              icon={Clear}
+              isIconOnly
+              aria-label="cancel"
```

Comment:
Since we are setting the `text` prop and the `isIconOnly` prop, we don't need to set the aria-label. The `<Button />` component [has internal logic to set an aria-label if both the `text` and `isIconOnly` props are used](https://github.com/cerner/terra-core/blob/master/packages/terra-button/src/Button.jsx#L79).

Reply:
Updated: [c5496c8](https://github.com/cerner/terra-core/pull/2317/commits/c5496c8d26b6ca995c3a7213a493c1d6169b626b)